pkg/resources/statefulsets/log.go

```diff
@@ -32,17 +32,33 @@ func logSelector(t *miniov1.Tenant) *metav1.LabelSelector {
 	}
 }
 
-// logMetadata returns the object metadata for Log pods
-func logMetadata(t *miniov1.Tenant) metav1.ObjectMeta {
+// logDbMetadata returns the object metadata for Log pods
+func logDbMetadata(t *miniov1.Tenant) metav1.ObjectMeta {
 	labels := make(map[string]string)
-	labels[miniov1.TenantLabel] = t.Name
+	labels[miniov1.LogDbLabel] = t.Name
 	for k, v := range t.LogPgPodLabels() {
 		labels[k] = v
 	}
 
-	return metav1.ObjectMeta{
+	meta := metav1.ObjectMeta{
 		Labels: labels,
 	}
+
+	if t.Spec.Log.Db != nil {
+		// attach any labels
+		for k, v := range t.Spec.Log.Db.Labels {
+			meta.Labels[k] = v
+		}
+		// attach any annotations
+		if len(t.Spec.Log.Db.Labels) > 0 {
```

Comment:
Shouldn't this be `if len(t.Spec.Log.Db.Annotations) > 0 {`?

Reply:
fixed